tests/test_schema.py

```diff
@@ -999,6 +999,48 @@ def test_load_schema_union_names():
     assert loaded_schema == expected_schema
 
 
+def test_load_schema_accepts_custom_repository():
+    class LocalSchemaRepository:
```

Comment:
I know it's not required, but do you think this should inherit from `AbstractSchemaRepository`?

Reply:
Yes that would be correct